apps/remote_control/test/lexical/remote_control/search/indexer/extractors/variable_test.exs

```diff
@@ -0,0 +1,603 @@
+        {:ok, [def, ref], doc} =
+          ~q[
+            #{unquote(def_type)} my_fun(%{key: var}) do
+              %{key: var}
+            end
+          ]
+          |> index()
+
+        assert_definition(def, :var)
+        assert_reference(ref, :var)
+
+        assert decorate(doc, def.range) =~ "#{unquote(def_type)} my_fun(%{key: «var»}) do"
+        assert decorate(doc, ref.range) =~ "  %{key: «var»}"
+      end
+
+      test "when it's a struct module" do
+        {:ok, [def, ref], doc} =
+          ~q[
+            #{unquote(def_type)} my_fun(%{key: var}) do
+              %{key: var}
+            end
+          ]
+          |> index()
+
+        assert_definition(def, :var)
+        assert_reference(ref, :var)
+
+        assert decorate(doc, def.range) =~ "#{unquote(def_type)} my_fun(%{key: «var»}) do"
+        assert decorate(doc, ref.range) =~ "  %{key: «var»}"
+      end
+
+      test "when it's a tuple entry " do
+        {:ok, [def, ref], doc} =
+          ~q[
+            #{unquote(def_type)} my_fun({var}) do
+              {var}
+            end
+          ]
+          |> index()
+
+        assert_definition(def, :var)
+        assert_reference(ref, :var)
+
+        assert decorate(doc, def.range) =~ "#{unquote(def_type)} my_fun({«var»}) do"
+        assert decorate(doc, ref.range) =~ "  {«var»}"
+      end
+
+      test "when it utilizes a pin " do
+        {:ok, [first_def, second_def, first_pin, other_def, second_ref, other_ref], doc} =
+          ~q"
+            #{unquote(def_type)} my_fun({first, second}) do
+              [^first, other] = second
+              other
+            end
+          "
+          |> index()
+
+        assert_definition(first_def, :first)
+
+        assert decorate(doc, first_def.range) =~
+                 "#{unquote(def_type)} my_fun({«first», second}) do"
+
+        assert_definition(second_def, :second)
+
+        assert decorate(doc, second_def.range) =~
+                 "#{unquote(def_type)} my_fun({first, «second»}) do"
+
+        assert_reference(first_pin, :first)
+        assert decorate(doc, first_pin.range) =~ "  [^«first», other]"
+
+        assert_definition(other_def, :other)
+        assert decorate(doc, other_def.range) =~ "  [^first, «other»]"
+
+        assert_reference(second_ref, :second)
+        assert decorate(doc, second_ref.range) =~ "  [^first, other] = «second»"
+
+        assert_reference(other_ref, :other)
+        assert decorate(doc, other_ref.range) =~ " «other»"
+      end
+    end
+  end
+
+  describe "variable definitions in anonymous function parameters are extracted" do
+    test "in a plain parameter" do
+      {:ok, [param], doc} =
+        ~q[
+         fn var ->
+          nil
+         end
+        ]
+        |> index_definitions()
+
+      assert_definition(param, :var)
+      assert decorate(doc, param.range) =~ "fn «var» ->"
+    end
+
+    test "in a struct's values" do
+      {:ok, [param], doc} =
+        ~q[
```

Comment:
The same issue, if I make some changes to this example, it fails to extract the reference.

```elixir
      {:ok, [param], doc} =
        ~q[
        fn %Pattern{foo: var} ->
          var
        end
        ]
        |> index_references()
```